### ISSUE TITLE
0.9 seem broken using tokio stream

### ISSUE THREAD

**Issue by Stargateur** [ebfull/pcap#187]
I update from 0.8 to 0.9 and just test today, my code capture one packet but nothing else.

git diff:
```rust
-    cap.filter("port sip").context(PacketCap)?;
+    cap.filter(&OPTIONS.filter, true).context(PacketCap)?;
```

I even try remove filter, nothing work expect go back to 0.8. If anyone can confirm ?

**Comment by DanielRussell** [comment]
Hello @Stargateur.  I've not seen this problem before.  Is your code available, or do you have a minimal reproducible test case you can share?

Do you know if your application only stops receiving packets, or does it lock up entirely?

The [included example](https://github.com/ebfull/pcap/blob/master/examples/streamlisten.rs) seems to work fine.

**Comment by Stargateur** [comment]
I try to repro from the example end up that it's using `tokio::main` that trigger the problem:

```rust
#[tokio::main]
async fn main() {
    let stream = start_new_stream().await;

    let fut = stream.for_each(move |s| {
        println!("{:?}", s);
        futures::future::ready(())
    });
    fut.await;
}
```

I don't see why, it should work the same.

**Comment by DanielRussell** [comment]
Thank you for the hint!  I've made a failing test case and will see if I can track down what needs to be fixed.

**Comment by DanielRussell** [comment]
@Stargateur, can you please try the most recent `tokio-multithread` branch on my fork and let me know if it fixes your issues?

**Comment by Stargateur** [comment]
Look it's working, the example work, and my application to capture sip packet seem ok too. Thank for the fix.

**Comment by stappersg** [comment]
On Sat, Oct 02, 2021 at 03:17:11AM -0700, Antoine wrote:
>  ... my application to capture sip packet seem ok too. Thank for the fix.

And is that fix in this git repository?


It is OK to beat me up with the URL of the merge request that I missed.



Regards  Stappers

**Comment by DanielRussell** [comment]
@stappersg the previous fix was only in my fork as I waited for feedback from @Stargateur.  PR #189 contains the fix, and should be clean enough to merge

**Comment by Wojtek242** [comment]
I have merged @DanielRussell fix and pushed an update to `0.9.1` to crates.io
